arraysForDeletion.js

```diff
@@ -0,0 +1,246 @@
+            for (let j = 0;j<emptyTabCopy.length;j++){
+                if (emptyTabCopy[j]===leftEdgesInt[i]){
+                    emptyTabCopy.splice(emptyTabCopy.indexOf(emptyTabCopy[j]),1)
+                }
+                else if (emptyTabCopy[j]===rightEdgesInt[i]){
+                    emptyTabCopy.splice(emptyTabCopy.indexOf(emptyTabCopy[j]),1)
+                }
+            }
+        }
+        console.log(emptyTabCopy)
+
+        array1.push(emptyTabCopy[0])
+        for (let k =0;k<5;k++){                      // rechecking massive 5 times to make sure every number is correct (if not rechecked it might left some doubles)
+
+            array1 = array1.filter((item, index) => {          // deleting doubles
+                return array1.indexOf(item) === index
+            })
+            for (let i = 0;i<emptyTabCopy.length;i++){
+                for (let j=0;j<array1.length;j++){
+                    if (emptyTabCopy[i]===array1[j]+1 || emptyTabCopy[i]===array1[j]-board1.y || emptyTabCopy[i]===array1[j]-1 || emptyTabCopy[i]===array1[j]+board1.y){
+                        array1.push(emptyTabCopy[i])
+                    }
+
+
+                }
+            }
+        }
+
+        arrayLeft = arrayLeft.filter((item, index) => {
+            return arrayLeft.indexOf(item) === index
+        })
+        function tabsSeparation(tempArraySideSeparate,edges,array) {
+            for (let i = 0;i<array.length;i++){             // separating right and left empty fields and adding them into a different array
+                let limit = board1.x
+                if (array[i]===edges[edges.length-1]){
+                    limit = 0
+                }
+                let getDocumentNext = document.getElementById(board1.tableId[array[i]+limit])
+                tempArrayForSeparation.push(array[i])
+                if (getDocumentNext.innerHTML !== ' '){
+                    tempArraySideSeparate.push(tempArrayForSeparation)
+                    tempArrayForSeparation = []
+                }
+                else if (array[i]===edges[edges.length-1]){
+                    tempArraySideSeparate.push(tempArrayForSeparation)
+                    tempArrayForSeparation = []
+                }
+
+
+
+
+
+            }
+        }
+        tabsSeparation(tempArrayLeftSeparate,leftEdgesInt,arrayLeft)
+
+        let set = new Set(tempArrayLeftSeparate.map(JSON.stringify));    // deleting double arrays between two-dimensional arrays
+        tempArrayLeftSeparate = Array.from(set).map(JSON.parse);
+
+
+
+        arrayRight = arrayRight.filter((item, index) => {
+            return arrayRight.indexOf(item) === index
+        })
+        tabsSeparation(tempArrayRightSeparate,rightEdgesInt,arrayRight)
+
+        set = new Set(tempArrayRightSeparate.map(JSON.stringify));
+        tempArrayRightSeparate = Array.from(set).map(JSON.parse);
+
+        let arrayLeftFinal = []
+        function arraySep(inArray,outArray,side) {
```

Comment:
name function base on what the function is doing for example `showFieldsWithNoMinesAround` which will show all the fields around the fields which contains zero mines around. BTW you should have this function made using recursion in your logic.